pkg/services/command.go

```diff
@@ -1,58 +1,74 @@
+//go:generate mockgen -source=command.go -destination=mock_services/command.go
 //go:build !windows
 // +build !windows
 
 package services
 
 import (
 	"crypto/tls"
+	"errors"
 	"net"
 	"os/exec"
 
 	"github.com/ansible/receptor/pkg/logger"
 	"github.com/ansible/receptor/pkg/netceptor"
-	"github.com/ansible/receptor/pkg/utils"
 	"github.com/creack/pty"
 	"github.com/ghjm/cmdline"
 	"github.com/google/shlex"
 	"github.com/spf13/viper"
 )
 
-func runCommand(qc net.Conn, command string, logger *logger.ReceptorLogger) error {
+type NetCForCommandService interface {
+	GetLogger() *logger.ReceptorLogger
+	ListenAndAdvertise(service string, tlscfg *tls.Config, tags map[string]string) (*netceptor.Listener, error)
+}
+
+func runCommand(qc net.Conn, command string, logger *logger.ReceptorLogger, utilsLib UtilsLib) error {
+	// Note: shlex.Split does not return error for the empty string
 	args, err := shlex.Split(command)
 	if err != nil {
 		return err
 	}
+	if len(args) == 0 {
+		return errors.New("command is malformed")
+	}
 	cmd := exec.Command(args[0], args[1:]...)
 	tty, err := pty.Start(cmd)
 	if err != nil {
 		return err
 	}
-	utils.BridgeConns(tty, "external command", qc, "command service", logger)
+	utilsLib.BridgeConns(tty, "external command", qc, "command service", logger)
 
 	return nil
 }
 
 // CommandService listens on the Receptor network and runs a local command.
-func CommandService(s *netceptor.Netceptor, service string, tlscfg *tls.Config, command string) {
+func CommandService(s NetCForCommandService, service string, tlscfg *tls.Config, command string, utilsLib UtilsLib) {
+	if command == "" {
+		s.GetLogger().Error("error initializing command service: %s\n", errors.New("command not provided"))
```

Comment:
The Error() function in the logger already prefixes message, so we dont need to start our sentence with error. Also, as the logger can take any number of arguments, we dont need to create a new "errors.New()" error here, just the message will be enough 😄